### ISSUE TITLE
How to control the size of the heatmap section to make more space for the legends section?

### ISSUE THREAD

**Issue by muzier9** [DingWB/PyComplexHeatmap#100]
![image](https://github.com/user-attachments/assets/e49476e6-3b79-466f-9752-c5a1865b22aa)
Whether it is a single heatmap or a double heatmap, sometimes the space of legends does not fully display all the legends content.

I've also noticed this in the PyComplexHeatmap document example.
![image](https://github.com/user-attachments/assets/05859009-51fe-4271-abaa-69a4f6887c2b)

**Comment by DingWB** [comment]
Thanks for reporting this. It has already been fixed. 
Please re-install from github.

**Comment by muzier9** [comment]
No, there's no change. It can't work.

**Comment by DingWB** [comment]
It works on my side.
You need to uninstall and install again from github:
`pip uninstall -y PyComplexHeatmap && pip install git+https://github.com/DingWB/PyComplexHeatmap`

**Comment by DingWB** [comment]
You can also increase `legend_width` to assign more space for legend

**Comment by muzier9** [comment]
If legends had only one column, it would show up very well. If there are many legends in two columns, it will still be incomplete. I tried to set the legend_width, but it doesn't seem to work. Why?

This is the automatic legend width :
Collecting annotation legends..
Collecting annotation legends..
Estimated legend width: 35.102777777777774 mm
Display as shown :
![image](https://github.com/user-attachments/assets/b8dc3db8-6e93-4ae9-b1a0-80e5dc206c56)

Then I set the legend_width : 
cm_ax, legend_axes = composite(cmlist=cmlist, main=0, legend_hpad=2, legend_y=0.88, legend_gap=10, col_gap=0.1, With width_ratios=[1, 1.2], legend_width=40), 

I should have expanded the legend_width to 40, but The image hasn't changed at all.

**Comment by DingWB** [comment]
Can you share your code and data with me? @muzier9 
You can also send me simulated data so that I can reproduce this issue and fix it.

**Comment by DingWB** [comment]
I test with this example data, and it works (https://dingwb.github.io/PyComplexHeatmap/build/html/notebooks/composite_heatmaps.html)

Here is the result
![image](https://github.com/user-attachments/assets/e1588864-3aeb-4e0c-aaad-5f1cac3f79dc)


Can you try a larger `legend_width`? Such as 60 or 100?

**Comment by muzier9** [comment]
I have sent the code and test data to you by email. Can you check it? I changed the legend_width and it didn't work.

**Comment by muzier9** [comment]
I know what the problem is. The ide I used before is Pycharm, I changed it to the one you use Jupyter and tried it again, and it can be successfully modified. But I want to use Pycharm for window display graphics, do you have any good ideas?

**Comment by DingWB** [comment]
It is working. Nothing wrong with the code. You should learn how to adjust the parameter

![image](https://github.com/user-attachments/assets/f91f5b26-1797-4d37-92e7-28f34bfffcab)

**Comment by DingWB** [comment]
> I know what the problem is. The ide I used before is Pycharm, I changed it to the one you use Jupyter and tried it again, and it can be successfully modified. But I want to use Pycharm for window display graphics, do you have any good ideas?

Just save it to pdf and the PDF should be fine
```python
plt.savefig("test.pdf", bbox_inches='tight')
plt.show()
```

**Comment by muzier9** [comment]
Yes! The saved pdf displays very well. I also wonder if there is any way to make the window display complete like pdf?

**Comment by DingWB** [comment]
No idea. You can explore it or search for it on Google about the pycharm.

**Comment by muzier9** [comment]
Thank you very much. These are very useful to me.

**Comment by DingWB** [comment]
I just released a new version (1.8.1); you can install it using `pip`
Please help me test it; if you find any issues, please report them.

**Comment by muzier9** [comment]
OK. I will use the new version.
